File: src/main/java/net/etum/shattered/player/Knight.java
package net.etum.shattered.player;

public class Knight {


    private int ad; // Physical damage
    private int ap; // Magical damage
    private int armor; // Armor that reduces physical damage
    private int resistance; // Shield that reduces magical damage
    private int health; // Health points
    private int mana; // Points to use skills
    private int experience; // Points to determine the level of the knight

    // Default constructor
    public Knight() {
        this(0, 0, 0, 0, 20, 50, 0); // Calls the parameterized constructor with default values
    }

    // Parameterized constructor
    public Knight(int ad, int ap, int armor, int resistance, int health, int mana, int experience) {
        this.ad = ad;
        this.ap = ap;
        this.armor = armor;
        this.resistance = resistance;
        this.health = health;
        this.mana = mana;
        this.experience = experience;
    }



    // Getter and Setter for ad (physical damage)
    public int getAd() {
        return ad;
    }

    public void setAd(int ad) {
        this.ad = ad;
    }

    // Getter and Setter for ap (magical damage)
    public int getAp() {
        return ap;
    }

    public void setAp(int ap) {
        this.ap = ap;
    }

    // Getter and Setter for armor
    public int getArmor() {
        return armor;
    }

    public void setArmor(int armor) {
        this.armor = armor;
    }

    // Getter and Setter for resistance
    public int getResistance() {
        return resistance;
    }

    public void setResistance(int resistance) {
        this.resistance = resistance;
    }

    // Getter and Setter for health
    public int getHealth() {
        return health;
    }

    public void setHealth(int health) {
        this.health = health;
    }

    // Getter and Setter for mana
    public int getMana() {
        return mana;
    }

    public void setMana(int mana) {
        this.mana = mana;
    }

    // Getter and Setter for experience
    public int getExperience() {
        return experience;
    }

    public void setExperience(int experience) {
        this.experience = experience;
    }

    // Method to display the knight's statistics
    public void displayStats() {
        System.out.println("Knight Statistics:");
        System.out.println("Physical Damage (AD): " + ad);
        System.out.println("Magical Damage (AP): " + ap);
        System.out.println("Armor: " + armor);
        System.out.println("Resistance: " + resistance);
        System.out.println("Health: " + health);
        System.out.println("Mana: " + mana);
        System.out.println("Experience: " + experience);
    }
}

File: src/main/java/net/etum/shattered/player/Knight.kt
package net.etum.shattered.player

import org.bukkit.entity.Player

class Knight @JvmOverloads constructor(
    private var ad: Int = 3,          // Physical damage
    private var ap: Int = 0,          // Magical damage
    private var armor: Int = 0,       // Armor that reduces physical damage
    private var resistance: Int = 0,  // Shield that reduces magical damage
    private var health: Int = 20,     // Health points
    private var mana: Int = 50,       // Points to use skills
    private var experience: Int = 0   // Points to determine the level of the knight
) {

    // Getters and Setters
    fun getAD(): Int {
        return ad
    }

    fun setAD(value: Int) {
        ad = value
    }

    fun getAP(): Int {
        return ap
    }

    fun setAP(value: Int) {
        ap = value
    }

    fun getArmor(): Int {
        return armor
    }

    fun setArmor(value: Int) {
        armor = value
    }

    fun getResistance(): Int {
        return resistance
    }

    fun setResistance(value: Int) {
        resistance = value
    }

    fun getHealth(): Int {
        return health
    }

    fun setHealth(value: Int) {
        health = value
    }

    fun getMana(): Int {
        return mana
    }

    fun setMana(value: Int) {
        mana = value
    }

    fun getExperience(): Int {
        return experience
    }

    fun setExperience(value: Int) {
        experience = value
    }

    // Method to display the knight's statistics
    fun displayStats(player: Player) {
        player.sendMessage("🛡️ Knight Statistics 🛡️:")
        player.sendMessage("⚔️ Physical Damage (AD): $ad")
        player.sendMessage("🔮 Magical Damage (AP): $ap")
        player.sendMessage("🔰 Armor: $armor")
        player.sendMessage("🔱 Resistance: $resistance")
        player.sendMessage("❤️ Health: $health")
        player.sendMessage("💙 Mana: $mana")
        player.sendMessage("⭐ Experience: $experience")
    }

    // Method to increase Physical Damage (AD)
    fun increaseAD(amount: Int) {
        ad += amount
    }

    // Method to increase Magical Damage (AP)
    fun increaseAP(amount: Int) {
        ap += amount
    }

    // Method to use mana
    fun useMana(amount: Int) {
        mana -= amount
        if (mana < 0) {
            mana = 0
        }
    }

    // Method to gain experience
    fun gainExperience(amount: Int) {
        experience += amount
    }
}
